spec/serializer_spec.rb

```diff
@@ -0,0 +1,166 @@
+# frozen_string_literal: true
+
+# typed: false
+
+describe Redcord::Serializer do
+  let(:klass) do
+    Class.new(T::Struct) do
+      include Redcord::Base
+
+      attribute :a, Integer, index: true
+      attribute :b, String, index: true
+      attribute :c, Integer
+      attribute :d, T.nilable(Time), index: true
+      attribute :float, T.nilable(Float), index: true
+      attribute :boolean, T.nilable(T::Boolean), index: true
+
+      def self.name
+        'RedcordSpecModel'
+      end
+    end
+  end
+
+  before(:each) do
+    klass.establish_connection
+  end
+
+  it 'works with time types' do
+    now = Time.zone.now
+    allow(Time.zone).to receive(:now).and_return(now)
+    allow(Time).to receive(:now).and_return(now)
+
+    instance = klass.create!(a: 3, b: '3', c: 3)
+    another_instance = klass.find_by(id: instance.id)
+
+    expect(instance.d).to be_nil
+    expect(instance.created_at.to_i).to eq(now.to_i)
+    expect(instance.updated_at.to_i).to eq(now.to_i)
+    expect(instance.created_at.to_i).to eq(another_instance.created_at.to_i)
+    expect(instance.updated_at.to_i).to eq(another_instance.updated_at.to_i)
+  end
+
+  it 'works with boolean types' do
```

Comment:
What is the value of this test specifically? Is it just to check that a boolean value is serialized/deserialized correctly by the redis client?

Reply:
yep!